Fix incorrect egg-info location for modified package_dir in setup.py

selimb opened the pull request:
fixes #464 

Check for `egg-info` in `sys.path` instead of directory containing `setup.py`. Thanks to @RonnyPfannschmidt for the hint. 

Commits have been split TDD-style to show that the tests fail initially and then a fix is applied. Left some comments on the commits as well. 

If the PR is to be accepted, I certainly could squash them all together.

asottile commented:
Should `ast.literal_eval` be used instead of `eval`? I think you only expect a list of strings out of `sys.path`

The-Compiler commented:
Or just use json instead of using print/eval as a serialization tool :wink:

selimb commented:
@asottile I guess from a style point of view `ast.literal_eval` would be preferable.  Wasn't really concerned with security in this case. 

@The-Compiler Hah! IMHO JSON is a bit overkill in this case (and requires additional imports). Could do it if requested by others I suppose.

selimb commented:
Rebased `tox-dev/master`. Switched to `ast.literal_eval`. It seems that `tests/test_z_cmdline.py::test_alwayscopy` now fails. :(

In fact, I checked out `tox-dev/master` and the aforementioned test fails there too.

obestwalter commented:
Hi @selimb, in what context does it fail? Travis and AppVeyor seem happy with master and I just ran the test locally with success.

selimb commented:
Hey @obestwalter, so I tried running the test on my linux machine and it seems happy then, but on OSX I get an error...I copied the error [to this gist](https://gist.github.com/selimb/8e100d30117b52547cf927257b6678c7)
In any case, that is out of this PR's scope.

obestwalter commented:
o.k. it's not an actual bug either AFAICT - it looks like a permission problem on your machine: `Operation not permitted: '/private/var/folders/81/yw_sbkk14kn_6mg5rqdh4s7w0000gn/T/pytest-of-selimb/pytest-100/test_alwayscopy0/example123/.tox/python/lib/python2.7/config/config.c'")`

But you're right: it has nothing to do with this PR.

selimb commented:
Aaah I see. Could very well be. 

Is this PR good to go then? Again, let me know if you'd rather I squash all the commits into one -- the commit message would be identical to this PR's title I guess.

obestwalter commented:
> Is this PR good to go then?

Any takers for a review? I might get around to review it in detail next week otherwise, can't promise anything though.

selimb commented:
Great! Cheers.
